app/features/build_runner/build_runner.rb

```diff
@@ -129,6 +132,52 @@ def checkout_sha
     def pre_run_action
       logger.debug("Running pre_run_action in checkout_sha")
       self.checkout_sha
+      self.setup_build_specific_environment_variables
+    end
+
+    def setup_build_specific_environment_variables
+      self.environment_variables_set = []
+
+      # We try to follow the existing formats
+      # https://wiki.jenkins.io/display/JENKINS/Building+a+software+project
+      env_mapping = {
+        BUILD_NUMBER: current_build_number,
+        JOB_NAME: self.project.project_name,
+        WORKSPACE: self.project.local_repo_path,
+        GIT_URL: self.repo.git_config.git_url,
+        GIT_SHA: self.current_build.sha,
+        BUILD_URL: "https://fastlane.ci", # TODO: actually build the URL, we don't know our own host, right?
+        CI_NAME: "fastlane.ci"
+      }
+
+      if self.git_fork_config && self.git_fork_config.branch.to_s.length > 0
+        env_mapping[:GIT_BRANCH] = self.git_fork_config.branch # TODO: does this work?
```

Comment:
it's tricky because the git branch might be a branch on a fork, soooo I dunno.

Reply:
This is how Jenkins handles it

```
GIT_BRANCH - Name of the remote repository (defaults to origin), followed by name of the branch currently being used, e.g. "origin/master" or "origin/foo"
```

I think that makes sense for now to use `origin/[branch_name]`, and then we can think more about how this works for forks

Reply:
just saying `here is the branch` is not enough to know how to check it out if it's on a fork. I'm sure Jenkins handles this by updating the origin or something.

Reply:
Ahh yeah, so maybe you should include the actual `clone_url` or something